samza-test/src/test/java/org/apache/samza/test/processor/TestStreamProcessor.java

```diff
@@ -247,4 +220,70 @@ private void verifyNumMessages(String topic, int expectedNumMessages) {
 
     Assert.assertEquals(count, expectedNumMessages);
   }
+
+  private static class Mocks {
```

Comment:
It will be awesome if we can rename to something other than Mock. Sounds like you are trying to wrap the producer, consumer, processor and listener into a class. What are we benefitting from this? 

It sound like you want a stream processor factory for testing purposes. Producer and consumer are already in a util method. Perhaps you simply want to move initConsumer and initProducer as a private method in TestStreamProcessor, instead of tying it together with "Mocks". Please correct me if I am wrong here.

Reply:
+1 from my side too or can rename it to something other than `Mocks`

Reply:
Discussed offline. I will rename this to something meaningful instead of mocks. You are right in some sense this is a wrapper class to consolidate all the dependencies required to unit test stream processor.

I could have them as private methods and have test methods invoke them to get instances of producer & consumer. IMO, it isn't any different except the current implementation makes it slightly readable in terms of setting up dependencies for my test in one place.
Let me know if you feel strongly about moving them as private methods.